pkg/distributor/distributor.go

```diff
@@ -244,6 +264,36 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 
 	var lastPartialErr error
 
+	// Check the HA cluster and instance labels.
+	if d.cfg.HAReplicas && len(req.Timeseries) > 0 {
+		ts := req.Timeseries[0]
+		cluster, replica, err := findHALabels([]byte(d.cfg.TrackerConfig.ReplicaLabel), []byte(d.cfg.TrackerConfig.ClusterLabel), ts.Labels)
+		// If the sample doesn't have either HA label, accept it.
+		if cluster == "" && replica == "" {
+			goto acceptSample
+		}
+		// Right now removeHALabels will only return an error if it finds one but not both of the labels.
+		if err != nil {
+			if d.cfg.HARejectOne {
```

Comment:
When would we ever turn this on?  Perhaps if this was per-use, but right now I'd say just remove it.